### DISCUSSION TITLE
How does this extension work?

### DISCUSSION THREAD

**Discussion by StableInfo** [glucauze/sd-webui-faceswaplab#76] [Guide/Doc]
Hello, 
I am bit lost,
When I open the tab "Explore model" I see multiple lines that I don't understand that very well:
![image](https://github.com/glucauze/sd-webui-faceswaplab/assets/136593585/1745644f-90a2-4371-8f7d-3284888e2154)

Then the extract tab, I tried it 3 times and was able to extract the EYES from faces and from a full body image? What is the best way to make it work?

Ffinally what does "build" tab do? Same question for the checkbox:
![image](https://github.com/glucauze/sd-webui-faceswaplab/assets/136593585/2a61e9d4-8c09-4df4-8c8d-5ae0dae93102)

Not sure how to make all of this work.

Thanks

**Comment by glucauze** [comment]
Explore model is a geekery. It's for people interested in understanding the layers of the model. It's not really interesting in itself. I made this for myself to see the architecture of the model. People interested can also use [netron](https://netron.app/) for that.

Did you try to read the documentation to understand the options ? https://glucauze.github.io/sd-webui-faceswaplab/doc/ . I quote the doc : 

> the tool part allows you to:

> +    build one of the face checkpoints
> +  compare two faces, works the same way as the compute similarity
> + extract faces from images
> + explore the model (not very useful at the moment)
> + analyze a face : This will give the output of the insightface analysis model on the first face found.

> **Build Checkpoints: The FaceTools tab now allows creating checkpoints, which facilitate face reuse. When a checkpoint is used, it takes precedence over the reference image, and the reference source image is discarded.**

I quote the FAQ : 
>What is a face checkpoint?

> A face checkpoint is a saved embedding of a face, generated from multiple images. This is accomplished via the build tool located in the sd tab. The build tool blends all images dropped into the tab and saves the resulting embedding to a file.

> The primary advantage of face checkpoints is their size. An embedding is only around 2KB, meaning it’s lightweight and can be reused later without requiring additional calculations.

> Face checkpoints are saved as .safetensors files.

**Comment by glucauze** [comment]
Here are some gifs to explain (non cherry picked, just random pictures) : 

## Simple Usage (roop like)

This use codeformer on all faces (including non swapped)

[simple.webm](https://github.com/glucauze/sd-webui-faceswaplab/assets/137925069/de00b685-d441-44f9-bae3-71cd7abef113)

## Advanced options 

This is use to improve results. This use upscaling and codeformer only on swapped faces

[advanced.webm](https://github.com/glucauze/sd-webui-faceswaplab/assets/137925069/50630311-bd25-487f-871b-0a44eecd435d)

## Inpainting 

This add inpainting on faces : 

[inpainting.webm](https://github.com/glucauze/sd-webui-faceswaplab/assets/137925069/3d3508e9-5be4-4566-8c41-8301b2d08355)

## Build and use checkpoints : 

[build.webm](https://github.com/glucauze/sd-webui-faceswaplab/assets/137925069/e84e9a3c-840d-4536-9fbb-09ed256406d7)

**Comment by StableInfo** [comment]
Pretty NEat! Thank you. You should share this in the forum

**Comment by StableInfo** [comment]
in the subreddit I meant*

**Reply by glucauze** [reply]
I don't use reddit :)

**Reply by StableInfo** [reply]
Why this profil pic? lol

**Comment by schoenid** [comment]
I don't know, how to achive that.
I've tried with a picture of my own face, doing the settings from the above movie clips.
But there is not one picture with any similarity to me, not even vague ...
Is there any tutorial for this?

**Reply by StableInfo** [reply]
If you dont find one for this one, just google roop sd
